lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnSubmissionHelper.java

```diff
@@ -279,6 +282,13 @@ public void submit() throws IOException, YarnException {
     }
 
     this.yarnClient.submitApplication(applicationSubmissionContext);
+
+    if (this.isUnmanaged) {
+      // For Unmanaged AM mode, add a new app token to the
+      // current process so it can talk to the RM as an AM.
+      final Token<AMRMTokenIdentifier> token = this.yarnClient.getAMRMToken(this.applicationId);
+      this.tokenProvider.addTokens(UserCredentialSecurityTokenProvider.serializeToken(token));
```

Comment:
@jwang98052 there is a `UNMANAGED_DRIVER` parameter in `YarnClientConfiguration` module. We set this parameter to `true` if we want to use full-scale YARN driver configuration in unmanaged mode. We also set it to `true` by default in the `UnmanagedAmYarnDriverConfiguration` (change pending).

**P.S.** I am afraid we've strayed way off the actual purpose of this PR, which is about adding a necessary security token to the unmanaged AM process. Let's move our discussion to the mailing list, JIRA, or other relevant PRs and stick to the subject of the PR here.